File: LinkedListJava/Node.java

public class Node {
    //node for a linked list
	private String data;
	private Node next;
	
	public Node(String data, Node next){
		this.data = data;
		this.next = next;
	} // end constructor
	
	public void setData(String data){
		this.data = data;
	}
	
	public String getData(){
		return this.data;
	}
	
	public void setNext(Node next){
		this.next = next;
	}
	
	public Node getNext(){
		return this.next;
	}
}

File: LinkedListCPP/Node.cpp
#include "Node.h"

//using namespace std;

Node::Node(){
    Node::data = "";
    Node::next = NULL;
};

Node::Node(std::string data, Node * next)
{
    Node::data = data;
    Node::next = next;
};

Node::~Node()
{
    //may not need anything
};
